packages/apps/job-launcher/server/src/modules/webhook/webhook.service.ts

```
@@ -158,10 +160,7 @@ export class WebhookService {
     );
 
     if (!jobEntity) {
-      throw new ControlledError(
-        ErrorWebhook.InvalidEscrow,
-        HttpStatus.BAD_REQUEST,
-      );
+      throw new NotFoundError(ErrorWebhook.InvalidEscrow);
```

Same for this and above: it can be just `new Error` because these are called in async flows, not in http

---

Even though these methods are called in async flows, they are originated from the createWebhook endpoint, which is HTTP-based. I believe we should return the appropriate error, like ValidationError or InvalidEscrow, to ensure the calling system (in this case, the oracle, but could be anyone) can handle it properly. Using a generic Error would make it harder to distinguish the nature of the issue and could complicate debugging or error handling on the caller's side.